Character logic test

danpottshimself opened the pull request:
I have added a test for the characterLogic service which changes player number and makes turn when both are humans.
Checks to see if the current Player changes when the function is called and dependant on the currentPlayer.

This uses Mocks and subsequently I added an if statement within the mock file to test this functionality.

dpotts7 committed:
Created test layout

dpotts7 committed:
Proxy testing completed.
Checks the new game and make turn functions pass the correct data

dpotts7 committed:
Created Mock file.

dpotts7 committed:
Added test for humanLogic function.

keithbarrow commented on .unit-test/proxy-test.js:
```diff
@@ -1,25 +1,47 @@
+            proxyName;
+        beforeEach(function(){
+            module('Services.MyModule');
+            inject(function( $httpBackend, _Proxy_  ){
```
As per the other PR: _Proxy_ is incorrectly cased.

dpotts7 committed:
added variables all under each other with no repeats.

keithbarrow commented on .unit-test/human-logic-test.js:
```diff
@@ -0,0 +1,34 @@
+
+        it('Checks that the player switching changes from player 1 to 2 each turn.', function(){
+            mocks.CoreData.currentPlayer = mocks.CharacterLogic.humanLogic(1);
+            mocks.CoreData.currentPlayer.should.equal('2');
```
This doesn't execute the code you are testing

keithbarrow commented on .unit-test/human-logic-test.js:
```diff
@@ -0,0 +1,34 @@
+        });
+        it('Checks that the player switching changes from player 2 to 1 each turn.', function(){
+            mocks.CoreData.currentPlayer =  mocks.CharacterLogic.humanLogic(2);
+            mocks.CoreData.currentPlayer.should.equal('1');
```
This doesn't execute the code you are testing

keithbarrow commented on .unit-test/human-logic-test.js:
```diff
@@ -0,0 +1,34 @@
+                $provide.service('CoreData', mocks.CoreData);
+            });
+            sandbox = sinon.sandbox.create();
+            characterLogic = sinon.sandbox.mock(mocks.CharacterLogic);
```
Mock id probably the wrong type of observer - you probably want a spy or stub. As there is other stuff to fix - fix this too.

dpotts7 committed:
Removed functionality from the mocks.
Tested using executed code.
Reduced amount of code used.
Added spy to check if "makeTurn" has been called.

keithbarrow commented:
Closing: this has all been re-factored
